libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -1112,6 +1112,48 @@ def RemoveExternalIps(self) -> Dict[str, str]:
     # Return the deleted external IP address for future use
     return external_ip_addresses
 
+  def AbandonFromMIG(self, instance_group: str) -> None:
+    """Abandons the instance from the managed instance group.
+
+    Args:
+      instance_group (str): The instance group that this instance should
+          be abondoned from.
+
+    Raises:
+      errors.ResourceDeletionError: If the request did not succeed.
+    """
+
+    def RaiseException(exception: Exception, details: str) -> None:
+      msg = ('Unable to abandon {0:s} from managed '
+             'instance group {1:s}: {2:s}').format(
+        self.name,
+        instance_group,
+        details,
+      )
+      raise errors.ResourceDeletionError(msg, __name__) from exception
```

Comment:
why did you decide to wrap the error in ResourceDeletionError? My thoughts are if the error is due to some lacking permission or resource not found (wrong name) this will make the error message can be confusing

Reply:
I agree, I'll change it to `ResourceNotFoundError`.